packages/langium-cli/src/generator/highlighting/monarch-generator.ts

```diff
@@ -0,0 +1,476 @@
+ * @param grammar Langium grammar to use in generating this Monarch syntax highlighting file content
+ * @param config Langium Config to also use during generation
+ * @returns Generated Monarch syntax highlighting file content
+ */
+export function generateMonarch(grammar: langium.Grammar, config: LangiumLanguageConfig): string {
+
+    const symbols = getSymbols(grammar);
+    const regex = /[{}[\]()]/;
+    const operators = symbols.filter(s => !regex.test(s));
+
+    // build absract monarch grammar representation
+    const monarchGrammar: MonarchGrammar = {
+        languageDefinition: {
+            name:       config.id,  // identifier for generating the grammar export
+            keywords:   getKeywords(grammar),
+            operators,
+            symbols,
+            tokenPostfix: '.' + config.id, // category appended to all tokens
+        },
+        tokenizer: {
+            states: getTokenizerStates(grammar)
+        }
+    };
+
+    // return concrete monarch grammar representation
+    return prettyPrint(monarchGrammar);
+}
+
+/**
+ * Gets Monarch tokenizer states from a Langium grammar
+ * @param grammar Langium grammar to source tokenizer states from
+ * @returns Array of tokenizer states
+ */
+function getTokenizerStates(grammar: langium.Grammar): State[] {
+
+    // initial state, name is arbitrary, just needs to come first
+    const initialState: State = {
+        name: 'initial',
+        rules: getTerminalRules(grammar)
+    };
+
+    const whitespaceState: State = {
+        name: 'whitespace',
+        rules: getWhitespaceRules(grammar)
+    };
+
+    const commentState: State = {
+        name: 'comment',
+        rules: getCommentRules(grammar)
+    };
+
+    // order the following additional rules, to prevent
+    // comment sequences being classified as symbols
+
+    // add include for the whitespace state
+    initialState.rules.push(whitespaceState);
+
+    // add operator & symbol case handling
+    initialState.rules.push({
+        regex: '@symbols',
+        action: [
+            {
+                guard: '@operators',
+                action: { token: 'operator' }
+            },
+            // by default, leave the symbol alone
+            {
+                guard: '@default',
+                action: { token: '' }
+            }
+        ]
+    });
+
+    return [
+        initialState,
+        whitespaceState,
+        commentState
+    ];
+}
+
+/**
+ * Pretty prints a monarch grammar into it's concrete form, suitable for writing to a file
+ * @param monarchGrammar Grammar to pretty print
+ * @returns Monarch grammar in concrete form, suitable for writing to a file
+ */
+function prettyPrint(monarchGrammar: MonarchGrammar): string {
+    const name = monarchGrammar.languageDefinition.name;
+    return ([
+        `// Monarch syntax highlighting for the ${name} language.`,
+        `export const language${name} = {\n`,
```

Comment:
Suggestion: It might make sense to just make this a default export.

Reply:
For `prettyPrint`? If we do that we also need to export the `MonarchGrammar` interface as well, is that what you're thinking? Personally, figured it would be good to only expose the `generateMonarch` function, similar to what we do for TextMate.

Reply:
I mean the `export const language${name}` part. That way you don't need to specify a name for the element. Just use `export default { ... }`.